crates/term/lib.rs

```diff
@@ -31,6 +31,9 @@ use slog::{Level, OwnedKeyValueList};
 use slog_stream::Format as StreamFormat;
 use slog_stream::{Decorator, RecordDecorator, Streamer, AsyncStreamer};
 
+pub type TimestampSink<'a> = &'a mut FnMut(fmt::Arguments) -> io::Result<()>;
+pub type TimestampFn = Box<Fn(TimestampSink) -> io::Result<()> + Send + Sync>;
```

Comment:
That seems overly complicated. A function taking an argument that is function that you can call with `fmt::Arguments` ...

Also it will still require allocation - not directly, but it will. (more on it  later)